### PR TITLE
Enable support for AutoRandom in tidb sink

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve? 
When syncing data from tidb to tidb, if the primary key is AutoRandom, insert operations will fail because by default the db session to the sink does not allow writing to the AutoRandom key.

### What is changed and how it works?
Adding code to set the session veriable allow_auto_random_explicit_insert to true when creating a mysqlSink.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

### CODE REVIEW

File: cdc/sink/mysql.go [hunk 1]
```diff
@@ -240,6 +240,17 @@ func (s *mysqlSink) adjustSQLMode(ctx context.Context) error {
 	return errors.Trace(err)
 }
 
+// enableAutoRandom enables writing AutoRandom values to the sink by setting
+// the session variable "allow_auto_random_explicit_insert".
+func (s *mysqlSink) enableAutoRandom(ctx context.Context) error {
+	rows, err := s.db.QueryContext(ctx, fmt.Sprintf("SET allow_auto_random_explicit_insert = true;"))
```
Comment (overvenus) [inline cdc/sink/mysql.go:246]:
```suggestion
	rows, err := s.db.QueryContext(ctx, "SET allow_auto_random_explicit_insert = true;")
```

File: cdc/sink/mysql.go [hunk 2]
```diff
@@ -365,6 +377,13 @@ func newMySQLSink(ctx context.Context, sinkURI *url.URL, dsn *dmysql.Config, fil
 		}
 	}
 
+	if scheme == "tidb" {
```
Comment (overvenus) [inline cdc/sink/mysql.go:380]: What about other sinks?

File: cdc/sink/mysql.go [hunk 3]
```diff
@@ -254,14 +254,36 @@ var defaultParams = params{
...
+	if err != nil && err != sql.ErrNoRows {
+		return "", errors.Trace(err)
+	}
+
+	if err == nil && autoRandomInsertEnabled == "off" {
+		dsnCfg.Params["allow_auto_random_explicit_insert"] = "1"
+		log.Debug("Set allow_auto_random_explicit_insert to 1")
+	}
+
+	log.Info("Connection string:", zap.String("dsnStr", dsnCfg.FormatDSN()))
```
Comment (overvenus) [inline cdc/sink/mysql.go:286]: Could it leak password?

File: cdc/sink/mysql.go [hunk 4]
```diff
@@ -279,9 +301,10 @@ func newMySQLSink(ctx context.Context, sinkURI *url.URL, dsn *dmysql.Config, fil
 	tz := util.TimezoneFromCtx(ctx)
 
 	var dsnStr string
+	var scheme string
 	switch {
 	case sinkURI != nil:
-		scheme := strings.ToLower(sinkURI.Scheme)
+		scheme = strings.ToLower(sinkURI.Scheme)
```
Comment (overvenus) [inline cdc/sink/mysql.go:307]:
```suggestion
		scheme := strings.ToLower(sinkURI.Scheme)
```

File: cdc/sink/mysql.go [hunk 5]
```diff
@@ -279,9 +301,10 @@ func newMySQLSink(ctx context.Context, sinkURI *url.URL, dsn *dmysql.Config, fil
 	tz := util.TimezoneFromCtx(ctx)
 
 	var dsnStr string
+	var scheme string
```
Comment (overvenus) [inline cdc/sink/mysql.go:304]:
```suggestion
```

File: cdc/sink/mysql.go [hunk 6]
```diff
@@ -132,6 +133,7 @@ func (s *mysqlSink) EmitDDLEvent(ctx context.Context, ddl *model.DDLEvent) error
 		)
 		return nil
 	}
+	ddl.Query = binloginfo.AddSpecialComment(ddl.Query)
```
Comment (leoppro) [inline cdc/sink/mysql.go:136]: I think we should log the specially processed query

File: cdc/changefeed.go [hunk 7]
```diff
@@ -16,6 +16,7 @@ package cdc
 import (
 	"context"
 	"fmt"
+	"github.com/pingcap/tidb/sessionctx/binloginfo"
```
Comment (amyangfei) [inline cdc/changefeed.go:19]: Please follow the import rules
```
standard lib

third-party lib
```

or try to use tools like https://godoc.org/golang.org/x/tools/cmd/goimports

**Pull request merged**